### ISSUE TITLE
Project-local CSharpier for Rider plugin

### ISSUE THREAD

**Issue by commonquail** [belav/csharpier#541]
Thank you so much for CSharpier, and recently the Rider plugin!

For now, the Rider plugin works only with a globally installed CSharpier (confirmed).

Looking at [the command][url-cmd] the plugin runs -- `dotnet csharpier --version` -- it seems to me that should work transparently with project local installations just as well, but the plugin short-circuits.

Looking at [the process execution][url-exec], I wonder if `Runtime.getRuntime()` returns some working directory context other than the project directory. In fact, it seems very likely to me that it would. If instead of `exec(String)` we use `exec(String, String[], File)`, we can specify the subprocess working directory, and since we have access to at least a [`Project`][url-project] it seems like we should be able to locate the project directory.

Concretely, given a path to some project directory that is the repository and solution root with a `.config/dotnet-tools.json` tool manifest, this variant invocation will correctly run the project-local CSharpier:

```java
var projectDir = "...";
var proc = Runtime.getRuntime().exec(
        "dotnet csharpier --version",
        null,
        new java.io.File(projectDir));
```

I don't know which directory is the correct one. I know that [the tool manifest][url-tool] can be placed in other locations but I don't know how it then behaves. In the worst case, it sounds like full support requires searching through the whole tree for every `dotnet-tools.json` file, but I strongly suspect that a simplistic heuristic, evolved organically, starting from the repository root and/or solution root would be the optimal RoI.

[url-tool]: https://docs.microsoft.com/en-us/dotnet/core/tools/troubleshoot-usage-issues#local-tools
[url-project]: https://plugins.jetbrains.com/docs/intellij/project.html
[url-exec]: https://github.com/belav/csharpier/blob/fc1b4aa43f470646df4728107a840cc8dc091c4e/Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/CSharpierService.java#L58
[url-cmd]: https://github.com/belav/csharpier/blob/fc1b4aa43f470646df4728107a840cc8dc091c4e/Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/CSharpierService.java#L71

**Comment by belav** [comment]
From what I know, if you run `dotnet csharpier` from a directory, dotnet will search up through parent directories looking for the `.config/dotnet-tools.json` and then use the version of csharpier specified there. And if it doesn't find one, then it will use the global version.

My current plan is to do essentially the same thing. When a file is opened, figure out which version of csharpier should be used to format it by looking for the tool manifest file. I think using solution root would probably work for most cases, but it shouldn't be difficult to handle the edge cases.

I have this partially implemented in the vscode extension, but it runs `dotnet csharpier --version` from the directory of the current file instead of looking for a manifest file. I think looking for the manifest file will be more efficient. 

Once the vscode extension is updated then I can venture back into the world of java and get the rider plugin working the same way. It is the one I personally use, so I'd like to get it updated soon. Your comments will save me some time, so thank you!

**Comment by belav** [comment]
This was implemented in the 1.2.0 version of the plugin.
